### PR TITLE
Signup: Add an abtest for post signup landing page.

### PR DESCRIPTION
**Pull request by taggon**
This PR will add an abtest to encourage free plan users to upgrade their plan for better SEO.

See p5uIfZ-6N7-p2 for more detail.

## How to test

1. Set abtest _postSignupUpgradeScreen_ to _modified_.
2. Create a new website and follow the steps with free plan.
3. You should be able to see the new post signup screen like the following.

screen_shot_2017-08-10_at_23_05_45

### CODE REVIEW

**Comment by fditrapani** [comment]
@taggon I tested this out and it looks good! I however notice a minor glitch when I picked a paying plan instead.

**Comment by travisw** [comment]
I too noticed an issue when choosing a paid plan. This new screen will show for a moment while things load and then go to the checkout.

A few other notes:

Sometimes when clicking "Upgrade Plan & Get A Domain" I get an error page. Maybe this is just something with the dev environment. Refreshing the page then loads it fine.

Clicking the "View my Site" CTA took me to wordpress.com instead of my site. I'm not sure if that's also just an issue with my dev environment.

**Comment by fditrapani** [comment]
I should have pressed the buttons too :P. 

The visit site link didn't work for me but the upgrade button works fine for me — it took me to the plans page. I wonder if we should send them to the [domain search screen](/domains/add/) instead though? Could be another test perhaps? cc @cristelrossignol

**Comment by cristelrossignol** [comment]
The 'View my site' call-to-action didn't work for me as well!

If technically possible, the 'Upgrade plan & get a domain' call-to-action could be clickable at all time instead of grayed out while waiting. (https://cloudup.com/ctIJFXPYo5m)

**Comment by cristelrossignol** [comment]
@fditrapani 
I think at this stage I prefer sending them to plans. It's more of an 'easy purchase' compared to finding a domain which can take so much time!

**Comment by fditrapani** [comment]
> I think at this stage I prefer sending them to plans. It's more of an 'easy purchase' compared to finding a domain which can take so much time!

Good hypothesis. I'm happy to start off like this. We can try the other destination in another test :)

**Comment by taggon** [comment]
@fditrapani @travisw Thanks for the feedback! I fixed the buttons and the momentary new screen before checkout.

**Comment by taggon** [comment]
@cristelrossignol The screen component doesn't have the site address while the site is preparing. Since the Upgrade button needs to know url of user's new site to take users to _wordpress.com/plans/NEW_SITE_URL_, the button should be disabled before the site is ready. If you think the button should be working without the address, could you tell me where it should head for?

**Comment by cristelrossignol** [comment]
@taggon Got it! Let's leave as is in that case! Thank you for the explanation

**Comment by travisw** [comment]
LGTM 👍 

I'm still getting an error sometimes when clicking "Upgrade Plan & Get A Domain", but if it works for others I'll assume that's just my environment.
screen shot 2017-08-10 at 10 37 38 am

The look on smaller screens could maybe use some tweaking, but it all works.

screen shot 2017-08-11 at 10 02 41 am

**Comment by fditrapani** [comment]
The upgrade button works for me.

File: client/signup/processing-screen/index.jsx [hunk 1]
```diff
@@ -3,24 +3,51 @@
...
 	static propTypes = {
 		hasCartItems: PropTypes.bool.isRequired,
 		loginHandler: PropTypes.func,
 		steps: PropTypes.array.isRequired,
 		user: PropTypes.object,
+		signupProgress: PropTypes.array,
 	};
 
+	componentWillMount() {
+		this.state = {
```
Comment (klimeryk) [inline client/signup/processing-screen/index.jsx]: @taggon 👋 This line is causing a warning in the console:
```
index.jsx:97 Warning: SignupProcessingScreen.componentWillMount(): Assigning directly to this.state is deprecated (except inside a component's constructor). Use setState instead.
```
Could you change it to `setState` instead?

**Pull request merged**